### PR TITLE
More fault-tolerance in the Worker API.

### PR DESCRIPTION
**Pull request by olonho**

### CODE REVIEW

File: runtime/src/main/cpp/Exceptions.h [hunk 1]
```diff
@@ -55,10 +55,11 @@ void RUNTIME_NORETURN ThrowNotImplementedError();
 // Throws illegal character conversion exception (used in UTF8/UTF16 conversions).
 void RUNTIME_NORETURN ThrowIllegalCharacterConversionException();
 void RUNTIME_NORETURN ThrowIllegalArgumentException();
+void RUNTIME_NORETURN ThrowIllegalStateException();
```
Comment (SvyatoslavScherbina) [inline runtime/src/main/cpp/Exceptions.h]: `ThrowIllegalStateException` is not `void`, it returns an object reference.

File: runtime/src/main/kotlin/kotlin/native/concurrent/Worker.kt [hunk 2]
```diff
@@ -24,15 +24,24 @@ import kotlinx.cinterop.*
...
          * better to use non-blocking IO combined with more lightweight coroutines.
          */
-        public fun start(): Worker = Worker(startInternal())
+        public fun start(errorReporting: Boolean = true): Worker = Worker(startInternal(errorReporting))
+
+        /**
+         * Return current worker, if known, null otherwise, for example, in the main thread or in platform
+         * thread without an associated worker.
+         */
+        public val current: Worker? get() {
```
Comment (SvyatoslavScherbina) [inline runtime/src/main/kotlin/kotlin/native/concurrent/Worker.kt]: The documentation is not clear enough to make a conclusion that `Worker.current != null` only in a thread that was created with `Worker.start()`.

Comment (olonho) [reply runtime/src/main/kotlin/kotlin/native/concurrent/Worker.kt]: Somewhat reworded.

File: runtime/src/main/cpp/Exceptions.h (hunk 1 above)
Comment (olonho) [reply runtime/src/main/cpp/Exceptions.h]: Offline.

**Pull request merged**